Watchdog component

radek2s opened the pull request:
Base on the issue #2051 that was request by one of our customers we provide a new Vue.js component that is responsible for checking the Scada-LTS health and if everything is ok this component should send a heartbeat tick to the WatchDog server. Now this component is checking the network connectivity and the application response. User can extend the behaviour of that component by adding additional conditions that should be check to inform user about the application state. 

![image](https://user-images.githubusercontent.com/22638815/151125940-504bad2c-e032-459f-bb74-c49edfa2405d.png)

Using info button user can check which validation step caused error. 

### Configuration: 
Details how to configure that component can be found [here](https://github.com/SCADA-LTS/Scada-LTS/tree/feature/%232051_is_alive2/scadalts-ui/src/components/graphical_views/watchdog) in the component documentation.

Limraj commented on src/org/scada_lts/web/mvc/api/components/is_alive/TimeFromServer.java:
```diff
@@ -42,4 +47,89 @@
+                .orElse(request.getRemoteAddr()) ;
+        messageStream.writeBytes(remoteAddress + "|" + config.getMessage());
+        socket.close();
+        messageStream.close();
```
This allows you to close resources no matter what happens in the code.
```
        try (Socket socket = new Socket(config.getHost(), config.getPort())) {
            try (DataOutputStream messageStream = new DataOutputStream(socket.getOutputStream())) {
                messageStream.writeBytes("ping");
            }
        }
```

Limraj commented on src/org/scada_lts/web/mvc/api/components/is_alive/TimeFromServer.java:
```diff
@@ -42,4 +47,89 @@
+     * @return ResponseEntity with status message
+     */
+    @PostMapping(value = "/watchdog")
+    public ResponseEntity<String> notifyWatchdog(@RequestBody WatchDogConfig config, HttpServletRequest request) {
```
`@RequestBody(required = true)`

Then if the object is null it will automatically return 400 code, you do not need to add if in the handling method of this endpoint.

Limraj commented:
1. You cannot create more than one such HTML component in a view. I add two with the same and different id to the view, the second component appears only after I delete one of them.
2. The component seems to be working when I am killing the application process:
![MicrosoftTeams-image (3)](https://user-images.githubusercontent.com/35842300/151551951-4b19abbc-fdf5-43dd-9885-e4c93cde2a33.png)
3. After starting the application and without logging in to the browser, he gets: 
![MicrosoftTeams-image (4)](https://user-images.githubusercontent.com/35842300/151551953-ff8e9cbb-186c-470d-bc2d-bada10278e69.png)
4. After logging in, it returns to normal, without refreshing the page:
![MicrosoftTeams-image (5)](https://user-images.githubusercontent.com/35842300/151551957-d7530d38-2ee4-4314-b2db-ec1c459cb980.png)
5. For datapoint set as alternate, binary and refresh period to 15s natomaist interval in component to 5s, for equals conditions and a value of true, it looks like if it hits 1, it shows a green "pass" message, while if it hits 0, it appears turns red so it looks okay:
![Screenshot 2022-01-28 at 14 17 31](https://user-images.githubusercontent.com/35842300/151553373-8da0ff86-09ea-46cc-8eb6-60bbfb897ad0.png)
![Screenshot 2022-01-28 at 14 17 47](https://user-images.githubusercontent.com/35842300/151553409-2306c5a6-910d-4c51-aa7d-b29fe0db8bd5.png)

6. For the binary point type, in the component for the "equals" condition we can enter true / false, we cannot enter the numeric value 1/0.
![Screenshot 2022-01-28 at 14 23 48](https://user-images.githubusercontent.com/35842300/151554306-c5c0ef32-200d-46e8-98b2-96c538e9e266.png)


7. To make netcat not stop the server after the first connection, you need to add the -k (keep) parameter, for example::
nc -4 -lk 127.0.0.1 3000

radek2s commented:
> 1. You cannot create more than one such HTML component in a view. I add two with the same and different id to the view, the second component appears only after I delete one of them.
>    edit:
>    An array of points may be entered, so this remark is not current:

Yes user  can provide a array of data points to check more than one DataPoint value.

> 2. The component seems to be working when I am killing the application process:
> 3. After starting the application and without logging in to the browser, he gets:
> 4. After logging in, it returns to normal, without refreshing the page:
> 5. For datapoint set as alternate, binary and refresh period to 15s natomaist interval in component to 5s, for equals conditions and a value of true, it looks like if it hits 1, it shows a green "pass" message, while if it hits 0, it appears turns red so it looks okay:

Great!

> 6. For the binary point type, in the component for the "equals" condition we can enter true / false, we cannot enter the numeric value 1/0.

Changed to numeric values instead of booleans

> 7. To make netcat not stop the server after the first connection, you need to add the -k (keep) parameter, for example::
>    nc -4 -lk 127.0.0.1 3000

Added to the documentation
